core/src/parser.rs

```diff
@@ -618,54 +636,125 @@ fn is_redacted(attrs: &[syn::Attribute]) -> bool {
     })
 }
 
-/// Check if we have a `target_os` cfg that dooes not match command line
-/// argument `--target-os`.
+/// Check if this attribute rejects our optional `target_os` arguments.
+///
+/// Examples:
+/// - `#[cfg(not(target_os = "windows"))]`
+/// - `#[cfg(not(any(target_os = "windows", target_os = "macos", feature = "my-feature")))]`
+/// - `#[cfg(not(all(target_os = "windows", feature = "my-feature")))]`
```

Comment:
This config feels a little tricky—We are essentially ignoring the other configurations that aren't `target_os`, however `not(all(target_os = "windows", feature = "my-feature"))` will evaluate to `true` even on Windows if the feature flag isn't set. So in that case, Typeshare would be ignoring the attached declaration and not generating types, even though the compiler would actually be including the declaration.

Where we're making assumptions about how to parse configs like this, it feels like we should update the docs in the `docs/` directory to clearly indicate how the option behaves so that users aren't surprised by things not working the way they expect.